lib/hawkular/operations/operations_api.rb

```diff
@@ -128,46 +128,113 @@ def invoke_specific_operation(operation_payload, operation_name, &callback)
       invoke_operation_helper(operation_payload, operation_name, &callback)
     end
 
-    # Deploys a war file into WildFly
+    # Deploys an archive file into WildFly
     #
     # @param [Hash] hash Arguments for deployment
     # @option hash [String]  :resource_path canonical path of the WildFly server into which we deploy
     # @option hash [String]  :destination_file_name resulting file name
     # @option hash [String]  :binary_content binary content representing the war file
-    # @option hash [String]  :enabled whether the deployment should be enabled or not
+    # @option hash [String]  :enabled whether the deployment should be enabled immediately, or not (default = true)
+    # @option hash [String]  :force_deploy whether to replace existing content or not (default = false)
+    # @option hash [String]  :server_groups comma-separated list of server groups for the operation (default = ignored)
```

Comment:
> upload stuff into the domain-controller content repo. 

This is doable by performing 'add_deployment(enabled=false)'

> when deploying to a server group, allowing to pick a deployment that already exists in the content repo of the DC. 

This is doable if you first perform the above, 'add_deployment(enabled=false)' and later perform an 'enable'

> (and with 1) also to the possibility to completely remove that content from the DC-content repo

This is doable by performing 'undeploy('removeContent=true)'.  Note that this is the default so really just performing 'undeploy' will do this.  You can call 'undeploy('removeContent=false)', although that is really a verbose way of just performing a 'disable'.

Note that there is one more option, add_deployment(forceDeploy).  If true content is replaced if it exists, if false, the default, an error is generated.